_posts/2019-04-29-opt-cuda-quantized.md

```diff
@@ -74,52 +76,68 @@ Figure 3. 2D convolution with data layout in NCHW4c and weight layout in OIHW4o4
-Finally, we build the model and run inference in the quantized mode.
+```python
+net = relay.quantize.quantize(sym, params=params)
+```
+
+Then, we use AutoTVM to extract tuning tasks for the operators in the model and perform automatic optimization. The [AutoTVM tutorial](https://docs.tvm.ai/tutorials/autotvm/tune_relay_cuda.html) provides an example for this.
 
+Finally, we build the model and run inference in the quantized mode.
+```python
+with relay.build_config(opt_level=3):
+    graph, lib, params = relay.build(net, target)
+```
+The result of `relay.build` is a deployable library.
+We can either run inference [on the GPU](https://docs.tvm.ai/tutorials/frontend/from_mxnet.html#execute-the-portable-graph-on-tvm) directly or deploy [on the remote devices](https://docs.tvm.ai/tutorials/frontend/deploy_model_on_rasp.html#deploy-the-model-remotely-by-rpc) via RPC.
 
 # Benchmark
 To verify the performance of the quantized operators in TVM, we benchmark the performance of several popular network models including VGG-19, ResNet-50 and Inception V3.
-We also benchmark on DRN-C-26, ResNeXt-50 to show the performance of the dilated convolution, the group convolution.
-DCN-ResNet-101 from [Deformable ConvNets](https://github.com/msracver/Deformable-ConvNets) is provided as an example of less conventional models.
-We choose TensorRT as our baseline.
+We also benchmark on DRN-C-26, ResNeXt-50, and DCN-ResNet-101 from [Deformable ConvNets](https://github.com/msracver/Deformable-ConvNets) to show the performance of the emerging models, which contains less conventional operators such as dilated convolutions, group convolutions and deformable convolutions.
+We choose NVIDIA TensorRT as our baseline.
 The result of MXNet 1.4 + cuDNN 7.3 in float32 mode is reported to show the speed up of quantization.
 The experiments are conducted on NVIDIA GTX 1080.
 We report the inference time per image when running in batch size = 1 and 16.
 
 As shown in the Figure 1, TVM achieves up to 8x speedup using quantization.
-Compared with TensorRT, TVM achieves parity with TensorRT performance in most models.
-TVM currently lags behind in DRN-C-26, the dilated convolution model.
-We can continue to improve it.
+In standard CNN models such as VGG and ResNet, TVM achieves parity with the state-of-the-art results from TensorRT.
 
-From the result of DCN-ResNet-101, the model with deformable convolutions, we can see that TVM can also bring speedup to less conventional models.
-Results of TensorRT is not available because there is no official implementation of the deformable convolution.
+In the benchmark of the emerging models, TVM achieves impressive results.
+We obtain significant performance gain on ResNeXt and DCN-ResNet-101.
+Result of DCN-ResNet-101 of TensorRT is not available because there is no official implementation of the deformable convolution.
+From this we can learn that the automatic optimization in TVM makes it easy and flexible to deal with emerging workloads.
```

Comment:
[We show that automatic optimization... to support and optimize emerging workloads.]